### DISCUSSION TITLE
What cameras, or features, are necessary to make good use of Quagga?

### DISCUSSION THREAD

**Discussion by ericblade** [ericblade/quagga2#106] [Q&A]
After having tried many different USB cameras and Laptop cameras, I had come to a conclusion that attempting to use several years old hardware, or cheap hardware, was not a good way to use Quagga.  I have, however, always had pretty decent success with mobile phone cameras.  However, today I encountered a USB camera that works *fantastically* with Quagga, so I thought maybe we should start a thread listing hardware and features of that hardware, so we can figure out what is necessary to get a good bar code reading.

If you have a device that reads a barcode decently, please comment it here, i'll add it to the list, we'll eventually make it a Wiki page, I guess, and add a link in the docs.

# USB Cameras known to work with Quagga
#################################
- Hewlett Packard Webcam HD-3110 https://amzn.to/2QKNu81
  720p CMOS sensor, Auto-Focus, 5.7Mpixel res
- Logitech QuickCam Orbit AF https://amzn.to/32REWly
  2MPixel (1600x1200 native) res, Auto-Focus, Motorized Tracking

# Built in Laptop cameras known to work with Quagga
########################################

- iMac 2017
- MacBook Pro 2012

# Mobile Phone cameras known to work with Quagga
########################################
- Pretty much everything built since 2014 or so seems to work pretty well.  Phones that default to using wide-angle lenses for their cameras will likely need to have a user select, or perform some automatic selection, of a non-wide-angle lense camera.  
- I have used Nexus 4, Nexus 5, and LG G6 regularly.
- iPad Pro 2018
- iPhone 6s

**Comment by ericblade** [comment]
Added Logitech QuickCam Orbit AF to known working list.  Works well attached to my laptop.  I'm leaning towards auto-focus being the big thing that makes cameras work well.

**Comment by kevinpapst** [comment]
I can confirm good rates (haven't tested all readers though) for the built in cameras of: 
iMac 2017, iPad Pro 2018, iPhone 6s, MacBook Pro 2012

**Comment by ericblade** [comment]
i'm guessing all of those have autofocus? I just added those to the top, thanks.  Perhaps we should make this a file in repo.

**Comment by kevinpapst** [comment]
Yes, I believe they have autofocus. And I think the `doc/` folder is the right place to put more infos  - I am not a fan of wiki, because this is not included in offline releases/clones and the README is already way too crowded (thats why I started the README PR).

**Comment by Ahedabu** [comment]
Hi,
It is supporting Zebra and Honeywell scanners?

**Reply by ericblade** [reply]
scanner? If you have a scanner, you probably don't need this software, this is for recognizing barcodes in images.
